src/configure/mod.rs

```diff
@@ -158,311 +103,312 @@ impl Context {
+    {
+        bail!("Account settings incomplete.");
+    }
 
-            if let Err(err) = imap.configure_folders(ctx, create_mvbox).await {
-                bail!("configuring folders failed: {:?}", err);
-            }
+    progress!(ctx, 600);
+    // try to connect to IMAP - if we did not got an autoconfig,
+    // do some further tries with different settings and username variations
+    try_imap_connections(ctx, param, param_autoconfig.is_some(), imap).await?;
 
-            if let Err(err) = imap.select_with_uidvalidity(ctx, "INBOX").await {
-                bail!("could not read INBOX status: {:?}", err);
-            }
-        }
-        17 => {
-            progress!(ctx, 910);
-            // configuration success - write back the configured parameters with the
-            // "configured_" prefix; also write the "configured"-flag */
-            // the trailing underscore is correct
-            param.save_to_database(ctx, "configured_").await?;
-            ctx.sql.set_raw_config_bool(ctx, "configured", true).await?;
-        }
-        18 => {
-            progress!(ctx, 920);
-            // we generate the keypair just now - we could also postpone this until the first message is sent, however,
-            // this may result in a unexpected and annoying delay when the user sends his very first message
-            // (~30 seconds on a Moto G4 play) and might looks as if message sending is always that slow.
-            e2ee::ensure_secret_key_exists(ctx).await?;
-            info!(ctx, "key generation completed");
-            progress!(ctx, 940);
-            return Ok(Step::Done);
-        }
-        _ => {
-            bail!("Internal error: step counter out of bound");
-        }
+    progress!(ctx, 800);
+    try_smtp_connections(ctx, param, param_autoconfig.is_some()).await?;
+
+    progress!(ctx, 900);
+
+    let create_mvbox = ctx.get_config_bool(Config::MvboxWatch).await
+        || ctx.get_config_bool(Config::MvboxMove).await;
+
+    if let Err(err) = imap.configure_folders(ctx, create_mvbox).await {
+        bail!("configuring folders failed: {:?}", err);
     }
 
-    Ok(Step::Continue)
+    if let Err(err) = imap.select_with_uidvalidity(ctx, "INBOX").await {
+        bail!("could not read INBOX status: {:?}", err);
+    }
+
+    progress!(ctx, 910);
+    // configuration success - write back the configured parameters with the
+    // "configured_" prefix; also write the "configured"-flag */
+    // the trailing underscore is correct
+    param.save_to_database(ctx, "configured_").await?;
+    ctx.sql.set_raw_config_bool(ctx, "configured", true).await?;
+
+    progress!(ctx, 920);
+    // we generate the keypair just now - we could also postpone this until the first message is sent, however,
+    // this may result in a unexpected and annoying delay when the user sends his very first message
+    // (~30 seconds on a Moto G4 play) and might looks as if message sending is always that slow.
```

Comment:
is this comment still correct? Refers to RSA i think and also the key is generated in paralllel, no?

Reply:
Key is not generated in parallel, we block here, waiting for the key to be generated. We could opportunistically start the key gen at the start of configure, and wait here for it to finish if we want to.

Reply:
i think, the comment can just be removed. at some time, key generation took loooooong time, i think today it is not worth any effort. it is fast.

Reply:
removed